src/pybind/mgr/cephadm/module.py

```diff
@@ -1045,7 +1045,7 @@ def remove_osds(self, osd_ids, services):
 
     def _create_daemon(self, daemon_type, daemon_id, host, keyring,
                        extra_args=None, extra_config=None,
-                       reconfig=False):
+                       reconfig=False, files=None):
```

Comment:
```suggestion
        def _create_daemon(self,
            daemon_type,  # type: str
            daemon_id,  # type: str 
            host,  # type: str 
            keyring,  # type: str
            extra_args=None,  # type: Optional[List[str]] 
            extra_config=None,  # type: ????
            reconfig=False,  # type: bool 
            files=None  # type: Optional[Dict[str, str]]
        ):
```
(that would be some form of documentation)

Reply:
How do we represent the files being either a Dict[str, str] or Dict[str, List[str]] ? Or should we just always assume the latter? Because they could be a combination. I use an isinstance(..., (list, tuple)). and turn it into 1 string in the list case.